src/job_worker/Worker.jl

```diff
@@ -0,0 +1,424 @@
+using Dates
+using Random
+using Logging
+using JSON3
+
+include("Config.jl")
+include("ECS.jl")
+include("HttpClient.jl")
+include("Jobs.jl")
+include("Storage.jl")
+
+"""
+Represents a job that needs to be processed
+"""
+struct Job
+    id::Int64
+    type::String
+    # We don't really know what type of data this will be 
+    input_payload::Any
+end
+
+"""
+Represents an assignment for a job
+"""
+struct JobAssignment
+    # ID of the assignment
+    id::Int64
+    # ID of the tasked job
+    job_id::Int64
+    # Path to where the data can be stored (in s3)
+    storage_uri::String
+end
+
+"""
+API Response after an assignment
+"""
+struct JobAssignmentResponse
+    assignment::JobAssignment
+end
+
+"""
+Context provided to job handlers with all necessary information
+"""
+struct JobContext
+    # The job to be processed
+    job::Job
+    # The job assignment details
+    assignment::JobAssignment
+    # The API client for making HTTP requests
+    http_client::Any
+    # Task metadata
+    task_metadata::Any
+
+    # Constructor that takes all fields
+    function JobContext(job, assignment, http_client, task_metadata)
+        return new(job, assignment, http_client, task_metadata)
+    end
+end
+
+"""
+Job handler type definition - a function that processes jobs of a specific type
+"""
+abstract type JobHandler end
+
+"""
+Default job handler that does minimal processing
+"""
+struct DefaultJobHandler <: JobHandler end
+
+"""
+Process a job with the default handler
+Return a tuple of (success::Bool, result_payload::Any)
+"""
+function process(::DefaultJobHandler, context::JobContext)
+    # Default implementation - 90% success rate
+    success = rand() > 0.1
+    sleep(rand(5:15))  # Simulate processing time between 5-15 seconds
+    return (success, success ? Dict() : nothing)
+end
```

Comment:
So these methods look useful for mock tests.

Suggest creating a `MockJobHandler` type and replacing the expected type for these methods (`DefaultJobHandler`) with that.

Reply:
Created a MockJobHandler to replace the JobHandler base - and added a mock argument to the service constructor which determines which handlers to register.